apps/etterlatte-behandling/src/main/kotlin/behandling/jobs/uttrekk/UttrekkLoependeYtelseEtter20JobService.kt

```diff
@@ -0,0 +1,118 @@
+import no.nav.etterlatte.Context
+import no.nav.etterlatte.Kontekst
+import no.nav.etterlatte.behandling.klienter.VedtakKlient
+import no.nav.etterlatte.funksjonsbrytere.FeatureToggle
+import no.nav.etterlatte.funksjonsbrytere.FeatureToggleService
+import no.nav.etterlatte.grunnlag.aldersovergang.AldersovergangService
+import no.nav.etterlatte.inTransaction
+import no.nav.etterlatte.libs.common.behandling.SakType
+import no.nav.etterlatte.libs.common.sak.SakId
+import no.nav.etterlatte.libs.ktor.token.HardkodaSystembruker
+import no.nav.etterlatte.sak.SakService
+import no.nav.etterlatte.vilkaarsvurdering.dao.VilkaarsvurderingDao
+import org.slf4j.LoggerFactory
+import java.time.LocalDate
+import java.time.Month
+import java.time.YearMonth
+
+enum class UttrekkToggles(
+    private val key: String,
+) : FeatureToggle {
+    UTTREKK_LOEPENDE_YTELSE_ETTER_20("uttrekk-loepende-ytelse-etter-20"),
+    ;
+
+    override fun key(): String = key
+}
+
+data class LoependeSak(
+    val sakId: SakId,
+    val foedselMnd: YearMonth,
+    val fyller20Mnd: YearMonth,
+    val opphoerDato: LocalDate,
+)
+
+class UttrekkLoependeYtelseEtter20JobService(
+    private val vedtakKlient: VedtakKlient,
+    private val sakService: SakService,
+    private val aldersovergangService: AldersovergangService,
+    private val vilkaarsvurderingDao: VilkaarsvurderingDao,
+    private val featureToggleService: FeatureToggleService,
+) {
+    private val logger = LoggerFactory.getLogger(this::class.java)
+
+    fun setupKontekstAndRun(context: Context) {
+        Kontekst.set(context)
+        run()
+    }
+
+    private fun run() {
+        if (featureToggleService.isEnabled(UttrekkToggles.UTTREKK_LOEPENDE_YTELSE_ETTER_20, false)) {
+            // Henter ut alle fødselsmåneder fra reformtidspunkt og frem til nå, hvor personer har blitt 20 år
+            // val foedselsmaanederSoekereFylt20 = hentMaanederSoekereFylt20()
+            // Begynner med et par hardkodede måneder før vi sjekker alle aktuelle måneder
+            val foedselsmaanederSoekereFylt20 = listOf(YearMonth.of(2004, Month.JANUARY), YearMonth.of(2004, Month.FEBRUARY))
+
+            val sakerSomSkulleVaertOpphoert = mutableListOf<LoependeSak>()
+
+            foedselsmaanederSoekereFylt20
+                .forEach { foedselsmaaned ->
+                    logger.info("Sjekker de som har blitt 20 år og ble født innenfor følgende måned: $foedselsmaaned")
+                    val sakIder = aldersovergangService.hentSoekereFoedtIEnGittMaaned(foedselsmaaned).map { SakId(it.toLong()) }
+
+                    sakIder
+                        // Ta bort de med yrkesskade-fordel. De skal opphøres først når de er 21 år
+                        .filterNot { sakId -> inTransaction { vilkaarsvurderingDao.hentMigrertYrkesskadefordel(sakId) } }
+                        .forEach { sakId ->
+                            try {
+                                val sak = inTransaction { sakService.finnSak(sakId) }
```

Comment:
Hent heller ut med spesifikk type?

```suggestion
                               val sak = inTransaction { sakService.finnSak(sakId, SakType.BARNEPENSJON) }
```

Reply:
har ikke ident på disse her så får ikke brukt den..